ccl/changefeedccl: TestChangefeedWithSimpleDistributionStrategy failed

ccl/changefeedccl.TestChangefeedWithSimpleDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/86dd5135-bf30-4e4c-8e32-482c0d40bc0d?testReportRun=9&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c561383c9d86a52cf63a2c8aaa3ee20270635f11](https://github.com/cockroachdb/cockroach/commits/c561383c9d86a52cf63a2c8aaa3ee20270635f11):


```
=== RUN   TestChangefeedWithSimpleDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedWithSimpleDistributionStrategy1311953055
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:345: found partitions: [{1 /Table/104/1{-/2}, /Table/104/1/{4-5}, /Table/104/1/{7-8}, /Table/104/1/{9-10}, /Table/104/1/1{3-4}, /Table/104/1/1{6-8}, /Table/104/1/{19-20}, /Table/104/1/2{6-7}, /Table/104/1/{28-31}, /Table/104/1/3{2-3} true 14} {4 /Table/104/1/{2-4}, /Table/104/1/{5-7}, /Table/104/1/{8-9}, /Table/104/1/1{0-1}, /Table/104/1/1{2-3}, /Table/104/1/1{5-6}, /Table/104/1/2{1-3}, /Table/104/1/5{8-9} true 11} {2 /Table/104/1/3{3-4}, /Table/104/1/3{6-7}, /Table/104/1/4{3-4}, /Table/104/1/4{6-8}, /Table/104/1/5{0-5}, /Table/104/1/{59-61}, /Table/104/{1/62-2} true 14} {6 /Table/104/1/{37-43}, /Table/104/1/4{4-6}, /Table/104/1/{48-50}, /Table/104/1/5{5-8}, /Table/104/1/6{1-2} true 14} {5 /Table/104/1/1{1-2}, /Table/104/1/1{4-5}, /Table/104/1/1{8-9}, /Table/104/1/2{0-1}, /Table/104/1/2{3-6}, /Table/104/1/2{7-8}, /Table/104/1/3{1-2}, /Table/104/1/3{4-6} true 11}]
    changefeed_dist_test.go:458: range counts: [14 14 0 11 11 14 0 0]
    changefeed_dist_test.go:518: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:518
        	Error:      	"14" is not less than or equal to "12"
        	Test:       	TestChangefeedWithSimpleDistributionStrategy
        	Messages:   	counts [14 14 0 11 11 14 0 0] contains value greater than upper bound 12
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedWithSimpleDistributionStrategy1311953055
--- FAIL: TestChangefeedWithSimpleDistributionStrategy (73.37s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedWithSimpleDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-36917

--- a/pkg/ccl/changefeedccl/changefeed_dist_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_dist_test.go
@@ -496,8 +496,6 @@ func TestChangefeedWithSimpleDistributionStrategy(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 120870)
-
 	// The test is slow and will time out under deadlock/race/stress.
 	skip.UnderShort(t)
 	skip.UnderDuress(t)
@@ -512,6 +510,8 @@ func TestChangefeedWithSimpleDistributionStrategy(t *testing.T) {
 	tester := newRangeDistributionTester(t, noLocality)
 	defer tester.cleanup()
 	tester.sqlDB.Exec(t, "SET CLUSTER SETTING changefeed.default_range_distribution_strategy = 'balanced_simple'")
+	// We need to disable the bulk oracle in order to ensure the leaseholder is selected.
+	tester.sqlDB.Exec(t, "SET CLUSTER SETTING changefeed.random_replica_selection.enabled = false")
 	tester.sqlDB.Exec(t, "CREATE CHANGEFEED FOR x INTO 'null://' WITH initial_scan='no'")
 	partitions := tester.getPartitions()
 	counts := tester.countRangesPerNode(partitions)

